hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/RowCustomColumnsSortPartitioner.java

```diff
@@ -51,7 +51,7 @@ public RowCustomColumnsSortPartitioner(String[] columnNames, HoodieWriteConfig c
   public Dataset<Row> repartitionRecords(Dataset<Row> records, int outputSparkPartitions) {
     return records
         .sort(Arrays.stream(sortColumnNames).map(Column::new).toArray(Column[]::new))
-        .coalesce(outputSparkPartitions);
+        .repartition(outputSparkPartitions);
```

Comment:
Should the `.coalesce` happen before `.sort` to control the parallelism? Is that the reason why the parallelism is not honored in the global sorting?